R/MixedModelsCommon.R

```diff
@@ -792,1293 +793,4580 @@
+                            type = "number")
+    tempTable$addColumnInfo(name = "se",
+                            title = gettext("SE"),
+                            type = "number")
+    tempTable$addColumnInfo(
+      name = "lowerCI",
+      title = gettext("Lower"),
+      type = "number",
+      overtitle = gettextf("%s%% CI", 100 * options$summaryCI)
+    )
+    tempTable$addColumnInfo(
+      name = "upperCI",
+      title = gettext("Upper"),
+      type = "number",
+      overtitle = gettextf("%s%% CI", 100 * options$summaryCI)
+    )
+    tempTable$addColumnInfo(name = "rhat",
+                            title = gettext("R-hat"),
+                            type = "number")
+    tempTable$addColumnInfo(name = "ess_bulk",
+                            title = gettext("ESS (bulk)"),
+                            type = "number")
+    tempTable$addColumnInfo(name = "ess_tail",
+                            title = gettext("ESS (tail)"),
+                            type = "number")
+
+    if (tableName == gettext("Model summary")) {
+      if(options$dependentVariable != "" &&
+         length(options$fixedVariables) > 0 &&
+         length(options$randomVariables) == 0) {
+        tempTable$addFootnote(.mmMessageMissingRE)
+      }
+      if (type == "BGLMM") {
+        if (options$family == "binomial_agg" &&
+            options$dependentVariableAggregation == "") {
+          tempTable$addFootnote(.mmMessageMissingAgg)
+        }
+      }
+
+      if(class(jaspResults[["mmModel"]]$object$model) %in% c("simpleError", "error")) {
+        STANOVAsummary$setError(gettext("The model could not be estimated. Please, check the options and dataset for errors."))
+      }
+      return()
+    }
+
+    for (j in 1:nrow(tempSummary)) {
+      tempRow <- list(
+        estimate   = tempSummary$Mean[j],
+        se         = tempSummary$MAD_SD[j],
+        lowerCI    = tempSummary[j, paste0((.50 - options$summaryCI / 2) *
+                                             100, "%")],
+        upperCI    = tempSummary[j, paste0((.50 + options$summaryCI / 2) *
+                                             100, "%")],
+        rhat       = tempSummary$rhat[j],
+        ess_bulk   = tempSummary$ess_bulk[j],
+        ess_tail   = tempSummary$ess_tail[j]
+      )
+
+      if (varName != "Intercept" && nrow(tempSummary) > 1) {
+        varName <-
+          paste(.unv(unlist(strsplit(
+            as.character(tempSummary$Variable[j]), ","
+          ))), collapse = jaspBase::interactionSymbol)
+        varName <- gsub(" ", "", varName, fixed = TRUE)
+        if (grepl(jaspBase::interactionSymbol, names(modelSummary)[i], fixed = T)) {
+          for (n in unlist(strsplit(.unv(names(
+            modelSummary
+          )[i]), jaspBase::interactionSymbol))) {
+            varName <- gsub(n, "", varName, fixed = TRUE)
+          }
+        } else{
+          varName <-
+            gsub(.unv(names(modelSummary)[i]), "", varName, fixed = TRUE)
+        }
```

Comment:
This part is a bit odd...

1. you're using `.unv`, but you really shouldn't.
2. a couple lines earlier you're doing `varName <- jaspBase::gsubInteractionSymbol(names(modelSummary)[i])`, so the names of the modelSummary don't contain the interaction symbol. But here you do `grepl(jaspBase::interactionSymbol, names(modelSummary)[i], fixed = T)`? Shouldn't that always be FALSE?
3. Please use `TRUE` instead of `T`.

Reply:
1) fixed
2) the previous lines extract the names and change them, the following lines refer back to the unchanged names (however, there should've been `:` instead of `jaspBase::interactionSymbol`, just fixed that)
The issue is that stanova produces variable names like this: `"facExperim control, facGender f"` and I want to:
a) change the "," to the interaction symbol
b) remove the variable names from the factor levels since I have the factor levels in the title.  
3) sure, fixed